mutexbot/reserve/action.yml

```diff
@@ -55,6 +57,13 @@ runs:
             handle_error "Unauthorized! Check your API keys."
           elif [ "$RESPONSE" -eq 404 ]; then
             handle_error "Resource not found."
+          elif [ "$RESPONSE" -eq 500 ]; then
+            if [ "$INTERNAL_SERVER_ERROR_COUNT" -eq 5 ]; then
+              handle_error "Too many internal server errors from the MutexBot API."
+            else
+              INTERNAL_SERVER_ERROR_COUNT=$((INTERNAL_SERVER_ERROR_COUNT + 1))
+              echo "Error: Internal server error from the MutexBot API. Retry ${INTERNAL_SERVER_ERROR_COUNT}/5."
+            fi
```

Comment:
We have `--retry` in the `curl` call — should we adjust it instead? Like increase the number of retries, tune timeouts? 

(and here just print an error message)

https://everything.curl.dev/usingcurl/downloads/retry.html#retry

Reply:
I missed that curl will have already retried this five times by the time we run into this case... In that case it probably makes sense to just increase the number of retries in curl directly and increase the time between retries.

wdyt about 10 retries instead of 5, and 2 seconds between instead of 1? This way their API has to be broken for 20 seconds instead of 5 before we see failures. Or maybe even more?

Reply:
> wdyt about 10 retries instead of 5, and 2 seconds between instead of 1?

Sure, let's try that. Let's bump it even to 30s (the total duration).

Another thought is that it might be us causing the 500 by hammering the same endpoint from different jobs 🤔